integrations/nvidia/src/haystack_integrations/components/rankers/nvidia/ranker.py

```diff
@@ -0,0 +1,204 @@
+import warnings
+from typing import Any, Dict, List, Literal, Optional
+
+from haystack import Document, component, default_from_dict, default_to_dict
+from haystack.utils import Secret, deserialize_secrets_inplace
+
+from haystack_integrations.utils.nvidia import NimBackend, url_validation
+
+_DEFAULT_MODEL = "nvidia/nv-rerankqa-mistral-4b-v3"
+
+_MODEL_ENDPOINT_MAP = {
+    "nvidia/nv-rerankqa-mistral-4b-v3": "https://ai.api.nvidia.com/v1/retrieval/nvidia/nv-rerankqa-mistral-4b-v3/reranking",
+}
+
+
+@component
+class NvidiaRanker:
+    """
+    A component for ranking documents using ranking models provided by
+    [NVIDIA NIMs](https://ai.nvidia.com).
+
+    Usage example:
+    ```python
+    from haystack_integrations.components.rankers.nvidia import NvidiaRanker
+    from haystack import Document
+    from haystack.utils import Secret
+
+    ranker = NvidiaRanker(
+        model="nvidia/nv-rerankqa-mistral-4b-v3",
+        api_key=Secret.from_env_var("NVIDIA_API_KEY"),
+    )
+    ranker.warm_up()
+
+    query = "What is the capital of Germany?"
+    documents = [
+        Document(content="Berlin is the capital of Germany."),
+        Document(content="The capital of Germany is Berlin."),
+        Document(content="Germany's capital is Berlin."),
+    ]
+
+    result = ranker.run(query, documents, top_k=2)
+    print(result["documents"])
+    ```
+    """
+
+    def __init__(
+        self,
+        model: Optional[str] = None,
+        truncate: Optional[Literal["NONE", "END"]] = None,
```

Comment:
Can we replace this with an enum like we've done in the [embedders](https://github.com/deepset-ai/haystack-core-integrations/blob/36c5401f2c8a3ae4465afcc00702e0d0f5cacdc7/integrations/nvidia/src/haystack_integrations/components/embedders/nvidia/truncate.py#L4)?

Reply:
i added `RankerTruncateMode(str, Enum)`

i did not use `from_str`, instead relying on `Enum` constructor. let me know if i should implement `from_str`.

Reply:
We'd prefer to keep it consistent across the codebase (it's a common pattern that we use), so it'd be great if we could add `from_str`, use `truncate: Optional[Union[RankerTruncateMode, str]] = None` as the init param and call `RankerTruncateMode.from_str` if the passed value is a string. You can basically use the same code as the one in the embedder truncate enum.